src/index.ts

```diff
@@ -12,13 +12,18 @@ import {
   existPages,
   existLocalesFolderWithNamespaces,
 } from './utils'
-import { LoaderOptions } from './types'
+import { LoaderOptions, NextConfigWithNextTranslate } from './types'
 import type { I18nConfig, NextI18nConfig } from 'next-translate'
 
 const test = /\.(tsx|ts|js|mjs|jsx)$/
 
-function nextTranslate(nextConfig: NextConfig = {}): NextConfig {
-  let basePath = pkgDir()
+function nextTranslate(
+  nextConfig: NextConfigWithNextTranslate = {}
+): NextConfig {
+  const config = nextConfig.nextTranslate ?? {}
+  delete nextConfig['nextTranslate']
+
+  let basePath = config.basePath ?? pkgDir()
```

Comment:
I recommend to don't mutate the parameter to avoid side-effects. We can solve it, for example, using destructuring:

```suggestion
function nextTranslate(
  config: NextConfigWithNextTranslate = {}
): NextConfig {
  const { nextTranslate: { basePath = pkgDir() } = {},  ...nextConfig } =  config ?? {}
 ```